src/pools/cluster/fixed.ts

```diff
@@ -0,0 +1,161 @@
+    public readonly numWorkers: number,
+    public readonly filePath: string,
+    public readonly opts: FixedClusterPoolOptions = { maxTasks: 1000 }
+  ) {
+    if (!cluster.isMaster) {
+      throw new Error('Cannot start a cluster pool from a worker!')
+    }
+    // TODO christopher 2021-02-09: Improve this check e.g. with a pattern or blank check
+    if (!this.filePath) {
+      throw new Error('Please specify a file with a worker implementation')
+    }
+
+    cluster.setupMaster({
+      exec: this.filePath
+    })
+
+    for (let i = 1; i <= this.numWorkers; i++) {
+      this.newWorker()
+    }
+  }
+
+  public destroy (): void {
+    for (const worker of this.workers) {
+      worker.kill()
+    }
+  }
+
+  /**
+   * Execute the task specified into the constructor with the data parameter.
+   *
+   * @param data The input for the task specified.
+   * @returns Promise that is resolved when the task is done.
+   */
+  public execute (data: Data): Promise<Response> {
+    // configure worker to handle message with the specified task
+    const worker: WorkerWithMessageChannel = this.chooseWorker()
+    // console.log('FixedClusterPool#execute choosen worker:', worker)
+    const previousWorkerIndex = this.tasks.get(worker)
+    if (previousWorkerIndex !== undefined) {
+      this.tasks.set(worker, previousWorkerIndex + 1)
+    } else {
+      throw Error('Worker could not be found in tasks map')
+    }
+    const id: number = ++this.id
+    const res: Promise<Response> = this.internalExecute(worker, id)
+    // console.log('FixedClusterPool#execute send data to worker:', worker)
+    worker.send({ data: data || {}, id: id })
+    return res
+  }
+
+  protected internalExecute (
+    worker: WorkerWithMessageChannel,
+    id: number
+  ): Promise<Response> {
+    return new Promise((resolve, reject) => {
+      const listener: (
+        message: MessageValue<Response>,
+        handle: SendHandle
+      ) => void = message => {
+        // console.log('FixedClusterPool#internalExecute listener:', message)
+        if (message.id === id) {
+          worker.removeListener('message', listener)
+          const previousWorkerIndex = this.tasks.get(worker)
+          if (previousWorkerIndex !== undefined) {
+            this.tasks.set(worker, previousWorkerIndex + 1)
+          } else {
+            throw Error('Worker could not be found in tasks map')
+          }
+          if (message.error) reject(message.error)
+          else resolve(message.data as Response)
+        }
+      }
+      worker.on('message', listener)
+    })
+  }
+
+  protected chooseWorker (): WorkerWithMessageChannel {
+    if (this.workers.length - 1 === this.nextWorker) {
+      this.nextWorker = 0
+      return this.workers[this.nextWorker]
+    } else {
+      this.nextWorker++
+      return this.workers[this.nextWorker]
+    }
+  }
+
+  protected newWorker (): WorkerWithMessageChannel {
+    const worker: WorkerWithMessageChannel = cluster.fork()
```

Comment:
Also here we could consider some options https://nodejs.org/dist/latest-v14.x/docs/api/cluster.html#cluster_cluster_fork_env

Reply:
Node cluster API is a little more complex than worker_theads

Reply:
I personally don't know what to do here. Please help me with that, or let us move that also into an new issue for later

Reply:
People needs error handling I am trying to draft something I could share a not working draft on another branch an let you know
This is really important to be done before to merge

Reply:
Oh okay
Could you explain (or share a link (preferable not the node docu, but a tutorial or blog post)) what this has todo with error handling and what this env is for?

Reply:
I apologies I think I confused the code with the error handling one.
Fork method has some parameters that we could add here that was my point

Reply:
fork method only has `env` as parameter :thinking:  
Sry when I'm confused, but I don't know what this "error-handling" does

Reply:
Nono my bad, don't care error handling in this point.

The point is to add the env parameter into the pool options( as optional) and then pass it  to the fork method, something like that

Reply:
Ok I, I will try to add this, but I need to be carefully in which branch I will make this, so I don't get to much conflicts

Reply:
`env` option added